src/Storefront/Controller/CartLineItemController.php

```diff
@@ -140,6 +162,24 @@ public function changeQuantity(Cart $cart, string $id, Request $request, SalesCh
         });
     }
 
+    #[Route(path: '/checkout/line-item/update', name: 'frontend.checkout.line-item.delete', defaults: ['XmlHttpRequest' => true], methods: ['POST', 'DELETE'])]
+    public function updateLineItems(Cart $cart, Request $request, SalesChannelContext $context): Response
+    {
+        return Profiler::trace('cart::update-line-items', function () use ($cart, $request, $context) {
+            try {
+                $this->cartItemUpdateRoute->change($request, $cart, $context);
```

Comment:
Don't pass the request directly to the store-api. Prepare the incoming "lineItems" like it is done in the `addLineItems` Method and pass that to a new Method `changeItemsQuantity` in the CartService.
```php
   /**
     * @throws CartException
     * @param array<string|int, array{id: string, quantity: int} $changeSet
     */
    public function changeItemsQuantity(Cart $cart, array $changeSet, SalesChannelContext $context): Cart
    {
        $request = new Request();
        $request->request->set('items', [
            [
                'id' => $identifier,
                'quantity' => $quantity,
            ],
        ]);

        $cart = $this->itemUpdateRoute->change($request, $cart, $context)->getCart();

        return $this->cart[$cart->getToken()] = $cart;
    }
```

Reply:
I created a corresponding `update` method in the `CartService` and used this method. Not sure if the change quantity method should be removed, or should stay, as it is a feature which is often needed.